### PR TITLE
[dubbo-1267]add consistentHashLoadBalance test 

### PR DESCRIPTION
**Pull request by dchack**
## What is the purpose of the change
add codes for test ConsistentHashLoadBalance.
fix [#1262](https://github.com/alibaba/dubbo/issues/1262)
## Brief changelog
1,add LoadBalanceTest.testConsistentHashLoadBalance_select for test ConsistentHashLoadBalance.
2,add LoadBalanceTest.randomString , generate random string for in invocation1.getArguments().
3,add ConsistentHashLoadBalance.NAME string.
## Verifying this change

run LoadBalanceTest.testConsistentHashLoadBalance_select

### CODE REVIEW

File: dubbo-cluster/src/main/java/com/alibaba/dubbo/rpc/cluster/loadbalance/ConsistentHashLoadBalance.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
  */
 public class ConsistentHashLoadBalance extends AbstractLoadBalance {
 
+    public static final String NAME = "consistenthash";
```
Comment (lovepoem) [inline dubbo-cluster/src/main/java/com/alibaba/dubbo/rpc/cluster/loadbalance/ConsistentHashLoadBalance.java]: Is this line you added is useless? @dchack

File: dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java [hunk 2]
```diff
@@ -117,6 +127,18 @@ public void testRoundRobinLoadBalance_select() {
...
 
+    @Test
+    public void testConsistentHashLoadBalance_select() {
+        Map<Invoker, AtomicLong> counter = getInvokeCounter(ConsistentHashLoadBalance.NAME, invocationList);
+
+        long runs = invocationList.size();
+        for (Invoker minvoker : counter.keySet()) {
+            Long count = counter.get(minvoker).get();
+            float ratio = count/(runs / (0f + invokers.size()));
+            Assert.assertTrue("invoker ratio must be gt 0.5 and lt 2",ratio>0.5 && ratio< 2);
```
Comment (chickenlj) [inline dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java]: Why `gt 0.5 and lt 2` is a reasonable ratio, can you explain?

File: dubbo-cluster/src/main/java/com/alibaba/dubbo/rpc/cluster/loadbalance/ConsistentHashLoadBalance.java (hunk 1 above)
Comment (dchack) [reply dubbo-cluster/src/main/java/com/alibaba/dubbo/rpc/cluster/loadbalance/ConsistentHashLoadBalance.java]: it's the same as RandomLoadBalance's codes.@lovepoem

Comment (diecui1202) [reply dubbo-cluster/src/main/java/com/alibaba/dubbo/rpc/cluster/loadbalance/ConsistentHashLoadBalance.java]: IMO, this is ok.

File: dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java [hunk 3]
```diff
@@ -71,6 +73,14 @@ public void setUp() throws Exception {
         invoker3 = EasyMock.createMock(Invoker.class);
         invoker4 = EasyMock.createMock(Invoker.class);
         invoker5 = EasyMock.createMock(Invoker.class);
+        invocationList = new ArrayList<Invocation>();
+        for(int i=0; i<1000; i++){
```
Comment (diecui1202) [inline dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java]: Could you format the code ?

File: dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java [hunk 4]
```diff
@@ -71,6 +73,14 @@ public void setUp() throws Exception {
...
+            Invocation  invocation1 = EasyMock.createMock(Invocation.class);
```
Comment (diecui1202) [inline dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java]: `invocation` is a better variable name here.

File: dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java (hunk 2 above)
Comment (diecui1202) [reply dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java]: I also have the same concern.

File: dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java [hunk 5]
```diff
@@ -71,6 +73,14 @@ public void setUp() throws Exception {
...
+            EasyMock.expect(invocation1.getMethodName()).andReturn("method1").anyTimes();
+            EasyMock.expect(invocation1.getArguments()).andReturn(new Object[]{randomString(3),randomString(5)}).anyTimes();
+            EasyMock.replay(invocation1);
```
Comment (diecui1202) [inline dubbo-cluster/src/test/java/com/alibaba/dubbo/rpc/cluster/loadbalance/LoadBalanceTest.java]: Here, we have changed test framework to Mockito, so could you change it to mockito again? Thanks.

**Comment by diecui1202** [comment]
@dchack Are you there? Now the master has changed a lot, so could you post a new PR with the advices above?

**Comment by chickenlj** [comment]
Thanks for the contribution. But i don't think we should merge this PR before the review passed. 
Reopen whenever necessary.

**Pull request closed without merging**